lib/Conversion/VerifToSV/VerifToSV.cpp

```diff
@@ -128,6 +128,128 @@ struct HasBeenResetConversion : public OpConversionPattern<HasBeenResetOp> {
   }
 };
 
+namespace VerifAssertLikeOp {
+
+static sv::EventControl verifToSVEventControl(verif::ClockEdge ce) {
+  switch (ce) {
+  case verif::ClockEdge::Pos:
+    return sv::EventControl::AtPosEdge;
+  case verif::ClockEdge::Neg:
+    return sv::EventControl::AtNegEdge;
+  case verif::ClockEdge::Both:
+    return sv::EventControl::AtEdge;
+  }
+  llvm_unreachable("Unknown event control kind");
+}
+
+// Convert the verif.assert like op that uses an enable, into an equivalent
+// sv.assert_property op that has the negated enable as its disable.
+template <typename TargetOp, typename Op, typename Adaptor>
+static LogicalResult convert(Op op, Adaptor operands,
+                             ConversionPatternRewriter &rewriter) {
+  // negate the enable if it exists
+  Value wdisable;
+  if (auto enable = operands.getEnable()) {
+    Value disable =
+        rewriter.createOrFold<comb::XorOp>(op.getLoc(), enable, enable);
+    wdisable = rewriter.createOrFold<hw::WireOp>(op.getLoc(), disable);
+  }
+
+  rewriter.replaceOpWithNewOp<TargetOp>(op, operands.getProperty(), wdisable,
+                                        operands.getLabelAttr());
+  return success();
+}
+
+template <typename TargetOp, typename Op, typename Adaptor>
+static LogicalResult convertClocked(Op op, Adaptor operands,
+                                    ConversionPatternRewriter &rewriter) {
+  // negate the enable if it exists
+  Value wdisable;
+  if (auto enable = operands.getEnable()) {
+    Value disable =
+        rewriter.createOrFold<comb::XorOp>(op.getLoc(), enable, enable);
+    wdisable = rewriter.createOrFold<hw::WireOp>(op.getLoc(), disable);
```

Comment:
I think a wire will be immediately folded away. We should insert a wire in PrepareForEmission.

Reply:
I think the right place to do this would be here https://github.com/llvm/circt/blob/961bad58f4f904f34b103b8072748e6441f53717/lib/Conversion/ExportVerilog/ExportVerilog.cpp#L784